smart-contracts/contracts/UnlockUtils.sol

```diff
@@ -0,0 +1,63 @@
+pragma solidity 0.5.7;
+
+// This contract provides some utility methods for use with the unlock protocol smart contracts.
+// Borrowed from:
+// https://github.com/oraclize/ethereum-api/blob/master/oraclizeAPI_0.5.sol#L943
+
+contract UnlockUtils {
+
+  function strConcat(
+    string memory _a,
+    string memory _b
+  ) internal
+    pure
+    returns(string memory)
+ {
+    bytes memory _ba = bytes(_a);
+    bytes memory _bb = bytes(_b);
+    string memory ab = new string(_ba.length + _bb.length);
+    bytes memory bab = bytes(ab);
+    uint k = 0;
+    for(uint i = 0; i < _ba.length; i++) {
+      bab[k++] = _ba[i];
+    }
+    for(uint i = 0; i < _bb.length; i++) {
+      bab[k++] = _bb[i];
+    }
+    return string(bab);
+  }
+
+  function strConcat(
+    string memory _a,
+    string memory _b
+  ) internal
+    pure
+    returns(string memory)
+  {
+    return strConcat(_a, _b, "", "", "");
+  }
+
+  function uint2str(
```

Comment:
Do we need this one?

Reply:
I think so. This is how it would be used in PublicLock:
```
function tokenURI(
    uint256 _tokenId
  ) external
    view
    isKey(_tokenId)
    returns(string memory)
  {
    string memory URI;
    if(baseTokenURI == '') {
      URI = unlockProtocol.getGlobalBaseURI;
    } else {
      URI = baseTokenURI;
    }
    return UnlockUtils.strConcat(
      URI,
      UnlockUtils.uint2str(_tokenId)
    );
  }
```

Reply:
maybe there's an easier way to convert uints to strings though.

Reply:
ah, makes sense.  lg